File: backend/app/schema/statistics.py
from pydantic import BaseModel
from typing import List
from .subscription import SubscriptionResponse

class CategoryStat(BaseModel):
    name: str
    icon: str
    total: float
    percent: float

class StatisticsResponse(BaseModel):
    total_monthly: float
    total_yearly: float
    active_count: int
    inactive_count: int
    top_subscriptions: List[SubscriptionResponse]
    category_stats: List[CategoryStat] = []
    currency: str = "RUB"

File: frontend/src/services/statistics.ts
import api from './api'
import { Subscription } from './subscriptions'

export interface Statistics {
    total_monthly: number
    total_yearly: number
    active_count: number
    inactive_count: number
    top_subscriptions: Subscription[]
    currency: string
}

/**
 * Get statistics for current user
 */
export async function getStatistics(): Promise<Statistics> {
    const response = await api.get<Statistics>('/statistics')
    return response.data
}
